yarn-project/acir-simulator/src/client/private_execution.test.ts

```diff
@@ -338,6 +350,99 @@ describe('Private Execution test suite', () => {
   });
 
+  describe('Consuming Messages', () => {
+    const contractDeploymentData = ContractDeploymentData.empty();
+    const txContext = new TxContext(false, false, false, contractDeploymentData);
+
+    let ownerPk: Buffer;
+    let owner: NoirPoint;
+    let recipientPk: Buffer;
+    let recipient: NoirPoint;
+
+    const buildMessage = async (content: Fr[], targetContract: AztecAddress, secret: Fr) => {
+      const wasm = await CircuitsWasm.get();
+
+      const contentBuf = Buffer.concat(content.map(field => field.toBuffer()));
+      const temp = toBigIntBE(sha256(contentBuf));
+      const contentHash = Fr.fromBuffer(toBufferBE(temp % Fr.MODULUS, 32));
+      console.log("content hash: ", contentHash.toBuffer().toString("hex"));
+
+      const secretHash = computeSecretMessageHash(wasm, secret);
+
+      return new L1ToL2Message(
+        new L1Actor(EthAddress.random(), 1),
+        new L2Actor(targetContract, 1),
+        contentHash,
+        secretHash,
+        0,
+        0,
+      );
+    };
+
+    beforeAll(() => {
+      ownerPk = Buffer.from('5e30a2f886b4b6a11aea03bf4910fbd5b24e61aa27ea4d05c393b3ab592a8d33', 'hex');
+      recipientPk = Buffer.from('0c9ed344548e8f9ba8aa3c9f8651eaa2853130f6c1e9c050ccf198f7ea18a7ec', 'hex');
+
+      const grumpkin = new Grumpkin(bbWasm);
+      owner = toPublicKey(ownerPk, grumpkin);
+      recipient = toPublicKey(recipientPk, grumpkin);
+    });
+
+    it('Should be able to consume a dummy cross chain message', async () => {
+      const db = levelup(createMemDown());
+      const pedersen = new Pedersen(bbWasm);
+
+      const contractAddress = AztecAddress.random();
+      const bridgedAmount = 100n;
+      const initialBalance = 0n;
+      const abi = NonNativeTokenContractAbi.functions.find(f => f.name === 'mint')!;
+
+      const secret = new Fr(1n);
+      // TODO: is just using the x coord for recipient ok?
+      const preimage = await buildMessage([new Fr(bridgedAmount), new Fr(recipient.x)], contractAddress, secret);
+
+      const messageKey = preimage.hash();
+
+      const tree: AppendOnlyTree = await newTree(
+        StandardTree,
+        db,
+        pedersen,
+        'l1ToL2Messages',
+        L1_TO_L2_MESSAGES_TREE_HEIGHT,
+      );
+
+      await tree.appendLeaves([messageKey.toBuffer()]);
+
+      const l1ToL2Root = Fr.fromBuffer(tree.getRoot(false));
+      const historicRoots = new PrivateHistoricTreeRoots(Fr.ZERO, Fr.ZERO, Fr.ZERO, l1ToL2Root, Fr.ZERO);
+
+      oracle.getL1ToL2Message.mockImplementation(async () => {
+        return Promise.resolve({
+          message: preimage.toFieldArray(),
+          index: 0n,
+          siblingPath: (await tree.getSiblingPath(0n, false)).data.map(buf => Fr.fromBuffer(buf)),
+        });
+      });
+
+      const txRequest = new TxRequest(
+        AztecAddress.random(),
+        contractAddress,
+        new FunctionData(Buffer.alloc(4), true, true),
+        // BUG: placing a fr in args will result in a fr wrapped in an fr
```

Comment:
have you flagged this issue somewhere?

Reply:
Ah I have not, but it is a known issue, https://github.com/AztecProtocol/aztec-packages/issues/611 made this for tracking